### ISSUE TITLE
tf.data.experimental.make_csv_dataset, currently only supports one label; However, most datasets can have more than one label

### ISSUE THREAD

**Issue by harica** [tensorflow/tensorflow#43083]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

test_dataset = tf.data.experimental.make_csv_dataset(
    test_fp,
    batch_size,
    column_names=column_names,
    label_name='species', M=<-- Currenly can only pass one label
    num_epochs=1,
    shuffle=False)

**System information**
- TensorFlow version (you are using): 2.3.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
tf.data.experimental.make_csv_dataset only accepts one label, to create the dataset. 

**Will this change the current api? How?**
API, needs to accept list of strings as label_name or a string containing delimiters for each column name as label_name

**Who will benefit with this feature?**
The community will benefit, by being able to generate more than one output label to the dataset, Otherwise, the API is useless for more than one output network
**Any Other info.**
See issue #43074 for sample test case.

**Comment by bhack** [comment]
The same feature was also requested at https://stackoverflow.com/questions/60713541/how-to-read-multiple-columns-as-labels-using-make-csv-dataset-in-tensorflow-2

**Comment by harica** [comment]
Above Stackoverflow proposed solution, referenced  beats the purpose why tf.data is being used, i.e. for tensorflow to handle the distributive data pre processing. If we use pd frame (in memory data from file), then distributive processing from csv files is lost, you are stuck with one node to process....

**Comment by aaudiber** [comment]
Supporting multiple labels seems reasonable to me. In the short term, you can achieve the same thing by re-organizing the data with a `map`. This just rearranges the Tensors, it doesn't require any copying or any expensive operations.

```
label_cols = ['label1', 'label2']
feature_cols = ['col1', 'col2']
test_dataset = tf.data.experimental.make_csv_dataset(
    test_fp,
    batch_size,
    column_names=label_cols+feature_cols
    num_epochs=1,
    shuffle=False)

def make_features_labels_tuple(x):
  labels = {k: x[k] for k in x if k in label_cols}
  features = {k: x[k] for k in x if k in feature_cols}
  return (features, labels)

test_dataset = test_dataset.map(lambda x: make_features_labels_tuple(x))
```

**Comment by candalfigomoro** [comment]
Related: https://github.com/tensorflow/tensorflow/issues/51571

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

By applying a map to reorganize the data, we can attain the same outcome without the need for data duplication or computationally intensive processes.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
